plugins/mergeStyles.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+exports.type = 'full';
+exports.active = true;
+exports.description = 'merge multiple style elements into one';
+
+var cssTools = require('../lib/css-tools');
+
+/**
+ * Merge multiple style elements into one.
+ *
+ * @param {Object} document document element
+ *
+ * @author strarsis <strarsis@gmail.com>
+ */
+exports.fn = function (document) {
+  // collect <style/>s (preserve order)
+  var styleEls = document.querySelectorAll('style');
+
+  // no <styles/>s, nothing to do
+  if (styleEls === null || styleEls.length <= 1) {
+    return document;
+  }
+
+  var styles = [];
+  for (var styleEl of styleEls) {
+    if (styleEl.isEmpty() || styleEl.closestElem('foreignObject')) {
+      // skip empty <style/>s or <foreignObject> content.
+      continue;
+    }
+
+    var cssStr = cssTools.getCssStr(styleEl);
+
+    styles.push({
+      styleEl: styleEl,
+
+      mq: styleEl.attr('media'),
+      cssStr: cssStr,
+    });
+  }
+
+  var collectedStyles = [];
+  var styleNo = 0;
+  for (var style of styles) {
```

Comment:
If you need current index use classic for loop instead

Reply:
Ah, the reason for this was that the linter complained about comparison of a string with a number in the `if (styleNo > 0) {` line, because the counter variable in `for ... of` loops are strings. I could use `parseInt(...)` in the if condition or disable the linter for that line.

Reply:
I mean
```js
for (let i = 0; i < styles.length; i += 1) {
  const style = styles[i]
}
```

Reply:
Yes, this is much better.